app/models/energy/construction.py

```diff
@@ -17,11 +17,18 @@ class WindowConstructionAbridged(BaseModel):
 
     name: str = Schema(
         ...,
-        regex=r'^[\s.A-Za-z0-9_-]*$',
         min_length=1,
         max_length=100
     )
 
+    @validator('name')
+    def check_name(cls, v):
+        assert all(ord(i) < 128 for i in v), 'Name contains non ASCII characters.'
+        assert all(char not in v for char in (',', ';', '!', '\n', '\t')), \
+            'Name constains invalid character for EnergyPlus (, ; ! \n \t).'
+        assert len(v) > 0, 'Name contains no valid characters.'
```

Comment:
This message is not accurate.  The words`valid characters` were used in my original code because we were automatically stripping out the invalid characters.  Now a message like `name contains no characters` or `name is an empty string` is more appropriate.